File: apps/web/src/types/meeting.ts
export type MeetingStatus =
  | "queued"
  | "processing"
  | "transcribed"
  | "analyzed"
  | "completed"
  | "failed";

export type Meeting = {
  id: string;
  title: string;
  source_type: "upload" | "youtube" | "microphone";
  language: string;
  status: MeetingStatus;
  progress: {
    stage: string;
    percent: number;
  };
  error_message: string | null;
  created_at: string;
  updated_at: string;
};

export type ActionItem = {
  title: string;
  owner: string | null;
  due_date: string | null;
  priority: "low" | "medium" | "high";
  context: string;
  confidence: number;
};

export type MeetingResult = {
  transcript: string;
  summary: string;
  decisions: string[];
  action_items: ActionItem[];
  risks: string[];
  follow_up_questions: string[];
};

File: apps/api/app/schemas/meeting.py
from datetime import datetime

from pydantic import BaseModel

from app.db.models import MeetingStatus, SourceType
from app.schemas.analysis import MeetingAnalysisPayload


class MeetingProgress(BaseModel):
    stage: str
    percent: int


class MeetingCreateResponse(BaseModel):
    id: str
    status: MeetingStatus
    title: str


class MeetingResponse(BaseModel):
    id: str
    title: str
    source_type: SourceType
    language: str
    status: MeetingStatus
    progress: MeetingProgress
    error_message: str | None = None
    created_at: datetime
    updated_at: datetime


class MeetingResultResponse(MeetingAnalysisPayload):
    transcript: str


class SlackSendRequest(BaseModel):
    webhook_url: str


class SlackSendResponse(BaseModel):
    ok: bool
    message: str
